src/components/atom/button/index.tsx

```diff
@@ -0,0 +1,22 @@
+import { ReactComponent as AddIcon } from "../../../assets/img/addIcon.svg";
+import "./styles.scss";
+
+type ButtonProps = {
+  text: string;
+  hasAddIcon?: boolean;
+  onClick: () => void;
+  type: HTMLButtonElement["type"];
+  disabled?: boolean;
+  styles: string;
+};
+
+const Button = ({ text, hasAddIcon, onClick, type, disabled, styles, ...rest }: ButtonProps) => {
+  return (
+    <button className={styles} onClick={onClick} disabled={disabled} type={type} {...rest}>
```

Comment:
Please use className instead of styles, because this component receives classes not styles.